xbmc/ServiceManager.cpp

```diff
@@ -188,6 +191,7 @@ void CServiceManager::DeinitStageTwo()
   m_binaryAddonManager.reset();
   m_addonMgr.reset();
   m_Platform.reset();
+  m_fileExtensionProvider.reset();
```

Comment:
This needs to go at the beginning of the function. Deinit in the opposite order as init to fix cross dependency issues

Reply:
Thx, will change the order